docs/migrations/chain-upgrade-guide-043.md

```diff
@@ -0,0 +1,233 @@
+
+*Note: In this example, we will be starting a new chain from `v0.42.6`. The binary for this version will be the genesis binary. For validators using Cosmovisor for the first time, the binary for the current version of the chain should be used as the genesis binary (i.e. the starting binary). For more information, see [Cosmovisor](../run-node/cosmovisor.html).*
+
+### Initial Setup
+
+From within the `cosmos-sdk` repository, check out `v0.42.6`:
+
+```
+git checkout v0.42.6
+```
+
+Build the `simd` binary for `v0.42.6` (the genesis binary):
+
+```
+make build
+```
+
+Reset `~/.simapp` (never do this in a production environment):
+
+```
+./build/simd unsafe-reset-all
+```
+
+Configure the `simd` binary for testing:
+
+```
+./build/simd config chain-id test
+./build/simd config keyring-backend test
+./build/simd config broadcast-mode block
+```
+
+Initialize the node and overwrite any previous genesis file (never do this in a production environment):
+
+<!-- TODO: init does not read chain-id from config -->
+
+```
+./build/simd init test --chain-id test --overwrite
+```
+
+Set the minimum gas price to `0stake` in `~/.simapp/config/app.toml`:
+
+```
+minimum-gas-prices = "0stake"
+```
+
+For the purpose of this demonstration, change `voting_period` in `genesis.json` to a reduced time of 20 seconds (`20s`):
+
+```
+cat <<< $(jq '.app_state.gov.voting_params.voting_period = "20s"' $HOME/.simapp/config/genesis.json) > $HOME/.simapp/config/genesis.json
+```
+
+Create a new key for the validator, then add a genesis account and transaction:
+
+<!-- TODO: add-genesis-account does not read keyring-backend from config -->
+<!-- TODO: gentx does not read chain-id from config -->
+
+```
+./build/simd keys add validator
+./build/simd add-genesis-account validator 5000000000stake --keyring-backend test
+./build/simd gentx validator 1000000stake --chain-id test
+./build/simd collect-gentxs
+```
+
+Now that our node is initialized and we are ready to start a new `simapp` chain, let's set up `cosmovisor` and the genesis binary.
+
+### Cosmovisor Setup
+
+First, install or update `cosmovisor`:
+
+```
+go get github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor
+```
+
+Set the required environment variables:
+
+```
+export DAEMON_NAME=simd
+export DAEMON_HOME=$HOME/.simapp
+```
+
+Set the optional environment variable to trigger an automatic restart:
+
+```
+export DAEMON_RESTART_AFTER_UPGRADE=true
+```
+
+Create the folder for the genesis binary and copy the `v0.42.6` binary:
+
+```
+mkdir -p $DAEMON_HOME/cosmovisor/genesis/bin
+cp ./build/simd $DAEMON_HOME/cosmovisor/genesis/bin
+```
+
+Now that `cosmovisor` is installed and the genesis binary has been added, let's add the upgrade handler to `simapp/app.go` and prepare the upgrade binary.
+
+### Chain Upgrade
+
+<!-- TODO: update example to use v0.43.0 -->
+
+Check out `release/v0.43.x`:
+
+```
+git checkout release/v0.43.x
+```
+
+Add the following to `simapp/app.go` starting on line 260:
```

Comment:
It seems to me these lines are not up-to-date anymore (L260 is about stakingKeeper). I think it's a bit risky to put explicit line numbers in this doc, as it's possible we'll backport stuff into app.go and forget to update this file.

```suggestion
Add the following to `simapp/app.go` inside `NewSimApp`:
```